lib/rules/jsx-key.js

```diff
@@ -61,18 +73,61 @@ module.exports = {
       return body.filter((item) => item.type === 'ReturnStatement')[0];
     }
 
-    return {
-      JSXElement(node) {
-        if (hasProp(node.openingElement.attributes, 'key')) {
-          return;
-        }
+    function nodeProcessed(node) {
```

Comment:
this reads like it's a predicate, but it also has the side effect of marking the node as processed. maybe it needs a better name?

Reply:
Yeah, you are right. 🤔 What do you think about `checkAndMarkNodeProcessed`? I am not crazy about this one but at least express better the behavior.